File: DSA/Hash/2215_Find_the_Difference_of_Two_Arrays.py
class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        nums1 = set(nums1)
        nums2 = set(nums2)
        diff_1 = nums1.difference(nums2)
        diff_2 = nums2.difference(nums1)
        return [list(diff_1),list(diff_2)]

File: DSA/Hash/2215_Find_the_Difference_of_Two_Arrays.go
package main

func findDifference(nums1 []int, nums2 []int) [][]int {
	a, b := uniq(nums1), uniq(nums2)
	var ans [][]int
	ans = make([][]int, 2)
	ans[0] = inOneNotInTwo(a, b)
	ans[1] = inOneNotInTwo(b, a)
	return ans
}

func uniq(x []int) map[int]bool {
	ans := make(map[int]bool)
	for _, v := range x {
		ans[v] = true
	}
	return ans
}

func inOneNotInTwo(a, b map[int]bool) []int {
	ans := make([]int, 0)
	for k, _ := range a {
		if _, ok := b[k]; !ok {
			ans = append(ans, k)
		}
	}
	return ans
}
